packages/plugin-degenfans-alfafrens/src/utils/degenfansApi.ts

```diff
@@ -0,0 +1,39 @@
+import { Staking } from "../types";
+
+
+export interface DegenFansResponse<T>{
+    status:number,
+    message:string,
+    data?:T
+}
+
+
+export interface StakingOption{
+  rank:number,
+  channelAddress:string,
+  name:string,
+  roi:number,
+  currentStake:number,
+}
+
+
+const degenfansApiBaseUrl="https://degenfans.xyz/servlet/rest-services/main/af/v1";
+ 
+  export function getStakingOptions(moxieId:string,wallets:string[], data:Staking): Promise<DegenFansResponse<StakingOption[]>> {
+    // For now, consider the data is stored on a static `users.json` file
+    return fetch(degenfansApiBaseUrl+'/alfafrens-staking-consultant/?token='+process.env.DEGENFANS_API, {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({moxieId,wallets, staking:data})
```

Comment:
Can this not forward `moxieId` to your server end? I think that'll be too sensitive and violate our policies

Reply:
I see 🤔 let me see actually I think we have utils function in moxie lib to get creator coin

might be better for you to do it that way

Other than `fid:` should also check creator coin that ends with `-fc`, they are the newer format coins that is connected to FC

Reply:
Yeah you should be able to use to get detail if user has any creator coin:
1. call [`getUserFtaData`](https://github.com/moxie-protocol/moxie-agent-skills/blob/fec58bdff6fff5ff761c581c8146e84dd9afe8b4/packages/moxie-agent-lib/src/services/fta.ts#L55)
2. If you need the address, you should be able to call from subgraph or [Token Manager](https://basescan.org/address/0xFC86e1dF300D1F1FA03C2d27C1815B36C3e396b1#readContract) contract and call `tokens` function, all you need is the `subjectAddress` from no. 1